app/scripts/modules/core/pipeline/config/stages/webhook/webhookStage.ts

```diff
@@ -78,14 +99,30 @@ export class WebhookStage {
       this.stage.customHeaders[customHeader.key] = customHeader.value;
     });
   }
+
+  public displayField(field: string): boolean {
+    return !this.preconfiguredProperties || !this.preconfiguredProperties.some(property => property === field);
+  }
+
+  public noUserConfigurableFields(): boolean {
+    if (this.preconfiguredProperties) {
+      if (this.viewState.waitForCompletion) {
+        return this.preconfiguredProperties.length >= 12;
```

Comment:
I moved the whole thing to Orca and just pass a boolean to Deck. It became much cleaner IMHO. https://github.com/spinnaker/orca/pull/1329/files#diff-bf5975fd9ad4864312a075624f079847R59